File: credit/credit.py
# import get_float function from cs50
from cs50 import get_float

# prompt the user to input a credit card number
while True:
    credit = get_float("Number: ")
    if credit > 0:
        break

# convert credit card number to a string
str_credit = str(int(credit))
# get the length of the string
length = len(str_credit)

sum = 0
# calculate the sum of every the other digit times 2, from the second-last digit
for i in range(length - 2, -1, -2):
    double_digit = 2 * int(str_credit[i])
    str_double_digit = str(double_digit)
    if len(str_double_digit) == 2:
        sum += int(str_double_digit[0]) + int(str_double_digit[1])
    else:
        sum += double_digit
# sum the other digit together
for j in range(length - 1, -1, -2):
    sum += int(str_credit[j])

# if the sum is dividable by 10
if (sum % 10 == 0):
    # VISA
    if (length == 13 or length == 16) and (str_credit[0] == '4'):
        print("VISA")
    # AMEX
    elif (length == 15) and ((str_credit[0] == '3') and (str_credit[1] == '4' or str_credit[1] == '7')):
        print("AMEX")
    # MASTERCARD
    elif (length == 16) and (str_credit[0] == '5' and (str_credit[1] == '1' or str_credit[1] == '2' or
                                                       str_credit[1] == '3' or str_credit[1] == '4' or str_credit[1] == '5')):
        print("MASTERCARD")
    else:
        print("INVALID")
else:
    print("INVALID")

File: credit/credit.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>
int main(void)

{
    //type a non-nagetive credit card number, without spaces or "-"
    long number;
    do
    {
      number = get_long_long("Number: ");
    }
    while (number < 0);

//    printf("%li", number);
    int sum = 0; //definine sum = every the other digit * 2 + left every the other digit
    int two_d; //digit * 2
    long digit1 = number; //transit number
    //sum of every the other digit * 2, starting the second to last digit
    for (int i = 0; digit1 > 100; i ++)
    {
        digit1 = number / pow(10, 2 * i + 1);
        two_d = (digit1 % 10) * 2;
        if (two_d < 10)
        {
            sum += two_d;
        }
        if (two_d >= 10)
        {
            sum += two_d % 10 + two_d / 10;
        }

//        printf("%i\n", sum);
    }
    //sum of every the other digit * 2, starting the second to last digit, plus the left digit
    long digit2 = number;
    for (int i = 0; digit2 > 10; i ++)
    {
        digit2 = number / pow(10, 2 * i);
        sum += digit2 % 10;
//        printf("%i\n", sum);
    }

//    printf("%i\n", sum);
    //count the length of number n, and get the first digit and first two digit
    long digit3 = number;
    int n = 0;
    int first_d = 0;
    int first_two_d = 0;

    while(digit3 != 0)
    {
        digit3 /= 10;
        ++ n;
        if (digit3 > 10 && digit3 < 100)
        {
            first_two_d = digit3;
        }
        if (digit3 > 1 && digit3 < 10)
        {
            first_d = digit3;
        }
    }
//    printf("%i\n", n);
//    printf("%i\n %i\n", first_two_d, first_d);
    //print VISA, AMEX,MASTERCARD or INVALID
    if (sum % 10 == 0)
    {
       if (first_d == 4 && (n == 13 || n == 16))
       {
           printf("VISA\n");
       }
       if ((first_two_d == 34 || first_two_d == 37) && (n == 15))
       {
           printf("AMEX\n");
       }
       if ((first_two_d == 51 || first_two_d == 52 || first_two_d == 53 || first_two_d == 54 || first_two_d == 55) && (n == 16))
       {
           printf("MASTERCARD\n");
       }
       else
       {
           printf("INVALID\n");
       }
    }
    else
    {
        printf("INVALID\n");
    }

}